Allow disabling one or more joysticks

rafaellehmkuhl opened the pull request:
This allows the user to disable some of their connected joysticks. This is needed when they have more than one joystick connected to their system, and they need to (e.g: one of them is used to controller another app), but they want only of them to control Cockpit (as having multiple controlling would lead to sticks alternating values).

https://github.com/user-attachments/assets/50243e87-4b24-4b00-997b-b400b42c92d9


Fix #1449.
To be merged after #1854.

ArturoManzoli reviewed (changes requested):
A few adjustments to the interface and usability (video):

(1) It's a little counterintuitive to have a switch set on the traditional off state, named `enabled`. You could do the inverse. Start with the switches on true labeled `enabled`, and the user can switch off the joystick he doesn't want to use.

(2) We can save some vertical space by moving the switch to the right part of the container:

(3) It would be nice to dimm the disabled joystick;

![image](https://github.com/user-attachments/assets/dcaf0793-5e32-4385-9b22-378a4b12d11b)


https://github.com/user-attachments/assets/9f1f55bd-8cd4-49d7-ae7a-385eacc1f38e

rafaellehmkuhl commented:
@ArturoManzoli I've reversed the enabling logic, so it becomes clear to the user (joysticks start enabled by default) and added the SVG dimm.

About moving the switch to the side to gain vertical space, I was actually able to gain it by removing some unnecessary vertical margin, and left the switch where it was, to become clear that the switch is for that joystick.

Updated the PR description with the current UI state.

ArturoManzoli commented:
> @ArturoManzoli can you check if this one is ready now?

Sure, just a sec

ArturoManzoli commented:
When at least one unknown joystick models are connected, only the table mode is available, and over there, there is no way to select and disable the secondary joysticks.

We have an open issue on this milestone do enable the visual joystick page for all joystick types, so this will not be a problem soon.
Should we add the switch also in table mode?

rafaellehmkuhl commented:
> When at least one unknown joystick models are connected, only the table mode is available, and over there, there is no way to select and disable the secondary joysticks.
> 
> We have an open issue on this milestone do enable the visual joystick page for all joystick types, so this will not be a problem soon. Should we add the switch also in table mode?
> 
> Imo it's ok to merge as it is

Nice catch. I will add the button to the table mode as well.

rafaellehmkuhl commented:
@ArturoManzoli fixed

ArturoManzoli commented:
The first controller title and switch are looking good, but from the second on, it is lacking a top margin. I think you can add a bottom margin on the buttons table so it will match the bottom margin on the page's horizontal toolbar.

![image](https://github.com/user-attachments/assets/426b5993-e841-4acc-a550-f7f13bbe58ec)

![image](https://github.com/user-attachments/assets/555e1527-fdab-4859-b94c-a2bd8cddac8e)

rafaellehmkuhl commented:
> The first controller title and switch are looking good, but from the second on, it is lacking a top margin. I think you can add a bottom margin on the buttons table so it will match the bottom margin on the page's horizontal toolbar.

Added.
